design/engineering-design.md

```diff
@@ -0,0 +1,27 @@
+# Engineering design
+
+This document seeks to convey how the commands flow through the code, and will highlight the public methods available.
+Additionally, the details of external resources will also be given.
+
+### The basic outline of the Moneymaker app is that a user can:
+- make a `transaction`
+- generate a `report` to view transactions
+
+#### An overview:
+The transactions are stored and retrieved to and from the user's folder on their local drive at the location `"~/.moneymaker"`. However, it is the filename that decides which account is being used to store transactions - therefor the system could (to be fulfilled on a future sprint) implement the possibility of allowing the user to choose the account, and view all historical transactions. At the moment, transactions can only be added to the file - there is no method of deleting transactions.  Furthermore, a report can be generated from these stored transactions for the user to view historic transactions and their balance. If no choice is made, or an erroneous choice is selected, then the application will print out a help file with the list of commands (to be implemented on a future sprint).
+
+## User requesting `transaction`:
+When `Main` is called, it starts the `Application` class, where a choice may be made to create a `transaction`. When a transaction is made by an `amount`, the application will infer via the presence of a negative sign if it is a deposit or withdrawal The transaction may include a `time` (in ISO 8601 date format [YYYY-MM-DD]) and a `description`.
```

Comment:
Feel free to reply to review comments that raises questions, or you disagree with. In this case it's because my review comment was too vague. Replying to your top level comment, we can be explicit about how the application knows it's a transaction command or a report command, by checking what the first argument is. If it's `transaction` it's a transaction command, if it's `report` it's a report command. It's borderline obvious, but this sort of rigour is required to ensure there's no ambiguity in regards to how the engineering design obeys the product design.